### DISCUSSION TITLE
"Has Anyone Completed the Full Course? Seeking Advice"

### DISCUSSION THREAD

**Discussion by abdullahansari-yk** [Cyfrin/foundry-full-course-cu#2663] [Q&A] [answered]
Ohkk!! 

"Imagine you're talking to your past self. What are some mistakes you would avoid if you were to take this course again?"

Please share your insights so I can learn from your experience tool!

**Comment by EngrPips** [comment, marked as answer]
Don't be in a rush to finish the course but instead be in a rush to understand everything in the course. The course provides everything you need to successfully learn it

**Reply by abdullahansari-yk** [reply]
How much time did you take to complete the course?

Just for rough estimation.

**Reply by EngrPips** [reply]
I studied the course over 3 times in the space of 8 months. I don't do any other thing than study, eat and sleep so you can understand why that was possible.

**Reply by Kinglucky04** [reply]
unfortunately I am a student, so i doubt that will be possible. but i am so locked in on this course!

**Reply by EngrPips** [reply]
Well, everyone can't do things the same way, and naturally things work out differently for everyone. It doesn't matter how long it takes what really matters is do you truly understand everything you have learned and can you do something meaningful with the knowledge.

**Reply by BowtiedHarpyEagle** [reply]
> How much time did you take to complete the course?
> 
> Just for rough estimation.

I have not yet completed the full course. I'm in the NFT section of the Advanced Foundry course. Right now I average about 5 hours of study for 1 hour of video course. I want to finish both this full course and Security course. Together they are about 60 hours of videos, so I schedule around 300 hours before I finish both. This is my current rough estimation. I think it will take me about 2 months of full time (40-50 hours a week) of study. 

But everyone is different, maybe you take more or less time to study than I do etc...

**Reply by EngrPips** [reply]
well said.
